src/modeler/AutoRest.Swagger/Validation/OperationIdNounPluralizedFormValidation.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class OperationIdNounPluralizedFormValidation : TypedRule<string>
```

Comment:
I'm not sure about the naming of this class, should be more like "OperationIdNounConflictWithModel" ? What the rule is trying to enforce is not about pluralizing the noun, it's about not conflicting with another name being used for another class name (class name of a model vs. class name of the one containing the operations).